pkg/syncer/metadatasyncer.go

```diff
@@ -201,7 +201,10 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 				log.Debugf("fsnotify event: %q", event.String())
 				if event.Op&fsnotify.Remove == fsnotify.Remove {
 					log.Infof("Reloading Configuration")
-					ReloadConfiguration(ctx, metadataSyncer)
+					if err := ReloadConfiguration(ctx, metadataSyncer); err != nil {
+						log.Errorf("failed to reload configuration. Current configuration unchanged.")
+						return
```

Comment:
What happens if this fails and `return`s? Is the container restarted or does this go func return, meaning that the container continues execution but subsequent changes to the configuration wont be caught by the container?

Reply:
I have removed `return` in the latest commit, and added a check to validate new VC credentials before re-instantiating VC.
```
// Verify if new configuration has valid credentials by connecting to vCenter.
				// Proceed only if the connection succeeds, else return error.
				newVC := &cnsvsphere.VirtualCenter{Config: newVCConfig}
				if err = newVC.Connect(ctx); err != nil {
					msg := fmt.Sprintf("failed to connect to VirtualCenter host: %s using new credentials, Err: %+v", newVCConfig.Host, err)
					log.Error(msg)
					return errors.New(msg)
				}
```

The subsequent changes will be captured even when the last attempt to change configuration failed.
And if new VC credentials are invalid, they will be ignored keeping in line with the current behavior.